packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -893,6 +920,65 @@ void main() {
     expect(camera.recorder!.qualitySelector, isNull);
   });
 
+  test(
+      'createCamera sets sensor and device orientations needed to correct preview rotation as expected',
+      () async {
+    final AndroidCameraCameraX camera = AndroidCameraCameraX();
+    const CameraLensDirection testLensDirection = CameraLensDirection.back;
+    const int testSensorOrientation = 270;
+    const CameraDescription testCameraDescription = CameraDescription(
+        name: 'cameraName',
+        lensDirection: testLensDirection,
+        sensorOrientation: testSensorOrientation);
+    const bool enableAudio = true;
+    const ResolutionPreset testResolutionPreset = ResolutionPreset.veryHigh;
+    const DeviceOrientation testUiOrientation = DeviceOrientation.portraitDown;
+    const DeviceOrientation testCurrentOrientation =
+        DeviceOrientation.portraitUp;
+
+    // Mock/Detached objects for (typically attached) objects created by
+    // createCamera.
+    final MockCamera mockCamera = MockCamera();
+    final MockProcessCameraProvider mockProcessCameraProvider =
+        MockProcessCameraProvider();
+    final MockCameraInfo mockCameraInfo = MockCameraInfo();
+    final TestSystemServicesHostApi mockSystemServicesApi =
+        MockTestSystemServicesHostApi();
+    TestSystemServicesHostApi.setup(mockSystemServicesApi);
+
+    // The proxy needed for this test is the same as testing resolution
+    // presets except for mocking the retrievall of the sensor and current
+    // UI orientation.
+    camera.proxy =
+        getProxyForTestingResolutionPreset(mockProcessCameraProvider);
+    camera.proxy.getSensorOrientation =
+        (_) async => Future<int>.value(testSensorOrientation);
+    camera.proxy.getUiOrientation =
+        () async => Future<DeviceOrientation>.value(testUiOrientation);
+
+    when(mockProcessCameraProvider.bindToLifecycle(any, any))
+        .thenAnswer((_) async => mockCamera);
+    when(mockCamera.getCameraInfo()).thenAnswer((_) async => mockCameraInfo);
+    when(mockCameraInfo.getCameraState())
+        .thenAnswer((_) async => MockLiveCameraState());
+
+    await camera.createCamera(testCameraDescription, testResolutionPreset,
+        enableAudio: enableAudio);
+
+    const DeviceOrientationChangedEvent testEvent =
+        DeviceOrientationChangedEvent(testCurrentOrientation);
+
+    DeviceOrientationManager.deviceOrientationChangedStreamController
+        .add(testEvent);
+
+    // Wait for currentDeviceOrientation to update.
+    await Future<void>.value();
```

Comment:
I couldn't figure out a great way to wait to check that the currentDeviceOrientation is received reliably, so if there is a better alternative let me know. Alternatively, I could move listening to changes in device orientation to a method and mock it, but was hoping to test it directly.

Reply:
I guess you could use `pumpEventQueue` or similar. It will end up doing the same thing.